src/ExpressReceiver.ts

```diff
@@ -57,10 +53,10 @@ export default class ExpressReceiver extends EventEmitter implements Receiver {
     }
   }
 
-  private requestHandler(req: Request, res: Response): void {
+  private async requestHandler(req: Request, res: Response): Promise<void> {
     let timer: NodeJS.Timer | undefined = setTimeout(
       () => {
-        this.emit('error', receiverAckTimeoutError(
+        this.bolt?.handleError(new ReceiverAuthenticityError(
           'An incoming event was not acknowledged before the timeout. ' +
           'Ensure that the ack() argument is called in your listeners.',
         ));
```

Comment:
> where the user already has no means to recover

They can if they listen for `ReceiverAckTimeoutError` errors and handle the how they need. (There is a bug there were I sent the wrong error, I'll fix that)

Slack's servers consider this a [failure condition](https://api.slack.com/events-api#failure_conditions) I would think sending an error would be the most appropriate thing.

Reply:
You set the timeout to `3001`, is that safe? `2800` is to give 200ms to respond, how much over 3s does slack serves wait?

Reply:
We're having trouble thinking of a use case where app developers can use `ReceiverAckTimeoutError` to recover. Do you have one in mind?

For example, my app could do something a little dangerous and say "if none of my listeners acked this, i'd like to ack it here in the global error handler, so that my end user doesn't have to see an error". That's probably not a good general rule, but maybe that's okay for some specific apps. Even in that case, the global error handler doesn't give you the context of the event that caused the error, or an `ack()` function to call. Without that context, you're not going to be able to do much, other than logging.

Reply:
AFAIK, its 3000ms + some magic constant. We know 3001 is not perfect at determining that a later `ack()` call would not work, but we know 2800 is even less perfect. We're just hoping to reduce the likelihood where you get a warning, then `ack()` is called and works in the nick of time. That situation can be very confusing.